Technical/issue 40 typescript babel configuration review

thescientist13 opened the pull request:
<!--
## Submitting a Pull Request
We love PRs and appreciate any help you can offer.

Please make sure the following criteria is meet before submitting your pull request.

1. <strong>PR meets the  [Contributing guidelines](./CONTRIBUTING.md)</strong>
2. Fork the repository and create your branch from <i>master</i>.
3. Ensure the test suite passes (`yarn run test`).
4. Make sure your code lints (`yarn run lint`).
-->

## Associated Issue
resolves #40 

## Changes Made
1.  Remove extra configs options from _tsconfig.json_
1.  Configure **babel-preset-env** in _.babelrc_ 
1.  Configure _webpack.config.common.js_ to use **babel-preset-env**
1.  Remove extra `devDependencies`
1.  Remove duplicate property from _jest.config.json_
1.  Have coverage run against "main" source code

ChrisPlease commented on webpack.config.common.js:
```diff
@@ -34,7 +34,7 @@ module.exports = {
         test: /\.tsx?$/,
         loaders: [
-          'babel-loader?presets[]=es2015',
+          'babel-loader?presets[]=env',
```
Why `atl` **and** `babel-loader`? It seems unnecessary. `awesome-typescript-loader` includes [first class integration with babel](https://www.npmjs.com/package/awesome-typescript-loader#differences-between-ts-loader). 

* [`useBabel` boolean](https://www.npmjs.com/package/awesome-typescript-loader#usebabel-boolean-defaultfalse) (defaulted to `false`), and
*  [`babelOptions` can pull from the `.babelrc` file](https://www.npmjs.com/package/awesome-typescript-loader#babeloptions-object-defaultnull). 

`babel-loader` seems unnecessary, unless there's something I'm missing?

thescientist13 replied:
@ChrisPlease 
Awesome, I didn't know that and was exactly what I was hoping to gather from our usage of TypeScript and Babel currently used.  Let me dig deeper into this, but this is definitely along the lines of what I was looking for.  (maybe **ast** will respect _.babelrc_ which would the 💣 )

thescientist13 commented:
@ChrisPlease  
Changes [made](https://github.com/kenzanlabs/react-seed/pull/50/commits/381c18b7a05996097d3186b5c2008861f930da9b) based on feedback
1.  removed **babel-loader** in _webpack.config.common.js_ from `.tsx` transpilation "pipeline"
1.  removed all JavaScript transpilation pipelines in _webpack.config.common.js_
1.  added **useBabel** and **useCache** options in _tsconfig.json_
1.  removed **babel-loader** and **babel-preset-es2015** from _package.json_

cc: @cxoltero / @thisgeek 

Thanks so much for reviewing, this is exactly what I was hoping to achieve 🎉 🍾
